spinalcordtoolbox/scripts/sct_register_to_template.py

```diff
@@ -369,13 +389,17 @@ def main(argv: Sequence[str]):
     printv('  Path template:        ' + path_template, verbose)
     printv('  Remove temp files:    ' + str(param.remove_temp_files), verbose)
 
+    # Compute center-of-mass of the rootlets segmentation:
+    if label_type == 'rootlet':
+        fname_rootlets_points = 'rootlets_mid.nii.gz'
+        fname_landmarks = sct_labels.cubic_to_point(Image(fname_rootlets)).save(fname_rootlets_points)
```

Comment:
Finding the rootlets center of mass might be strongly influenced by the quality of the rootlets segmentation. If we had some missing rootlets, for example, due to lower contrast, artifact, or canal stenosis, the center of mass might even be shifted outside the cord. Consider the following example where the left C2 rootlets are entirely missing (I removed it manually in FSLeyes), then the center of mass (the blue point is outside the cord).

<img width="1473" alt="image" src="https://github.com/user-attachments/assets/ca25138e-5b2b-419f-96c2-d2e5976fe348" />

Debugging:

```console
sct_label_utils -i sub-amu01_space-other_T2w_rootlets.nii.gz -cubic-to-point -o sub-amu01_space-other_T2w_rootlets_midpoints.nii.gz
sct_label_utils -i sub-amu01_space-other_T2w_rootlets_missing_left_C2.nii.gz -cubic-to-point -o sub-amu01_space-other_T2w_rootlets_missing_left_C2_midpoints.nii.gz
```

[sub-amu01_space-other_T2w_rootlets_missing_left_C2.nii.gz](https://github.com/user-attachments/files/18188861/sub-amu01_space-other_T2w_rootlets_missing_left_C2.nii.gz)
[sub-amu01_space-other_T2w_rootlets.nii.gz](https://github.com/user-attachments/files/18188862/sub-amu01_space-other_T2w_rootlets.nii.gz)

Reply:
To avoid this, perhaps we could consider centering the rootlets center of mass to the spinal cord centerline?

Reply:
hmm, I though I was projecting on the centerline, let me check!

Reply:
Ahhh, actually, all the landmarks are projected on the spinal cord segmentation:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/b4d8faeb6bfd79199a371dae70faf145c33618c5/spinalcordtoolbox/scripts/sct_register_to_template.py#L459-L462

Reply:
I see! I got probably confused by the `rootlets_mid.nii.gz` temp file:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/b4d8faeb6bfd79199a371dae70faf145c33618c5/spinalcordtoolbox/scripts/sct_register_to_template.py#L394-L396

which does _**not**_ have the landmarks projected on the spinal cord: 

<img width="1473" alt="image" src="https://github.com/user-attachments/assets/7e29a79a-964c-4355-b57b-e4791cab8477" />

Command I used:

```console
sct_register_to_template -i sub-amu01_space-other_T2w.nii.gz -s sub-amu01_space-other_T2w_seg.nii.gz -lrootlet sub-amu01_space-other_T2w_rootlets_missing_left_C2.nii.gz  -ofolder reg_rootelts_testing
```

Reply:
yes exaclty! I have a todo to remove this file to avoid confusion! or move to tmp